### PR TITLE
test images: Adds Windows Container images support (part 2)

### LINKED ISSUE #77268: Kubernetes test images should include Windows images in their manifest lists
**What would you like to be added**:

/sig windows
/sig testing

/area conformance

The images used in Kubernetes tests should also include Windows images in their manifest lists. Right now, in order to run E2E tests with Windows nodes, the env var ``KUBE_TEST_REPO_LIST`` needs to be set, containing alternative repositories which contain Windows images (the ``e2eteam`` repository is mainly used). The Windows images are currently maintained and built from ``kubernetes-sigs/windows-testing``: https://github.com/kubernetes-sigs/windows-testing/blob/master/images/BuildImageManifestLists.ps1

The build script in ``kubernetes/test/images`` does not build Windows images at the moment.

**Why is this needed**:

Running E2E tests on Windows nodes will be made easier, with the official Kubernetes test images.

Secondly, Hybrid OS tests will be able to be run.

### PR DESCRIPTION
**Pull request by claudiubelu**
**What type of PR is this?**

/kind feature

/sig testing
/sig windows

**What this PR does / why we need it**:

Some images require a different ``Dockerfile`` for Windows images, since they have different ways of  installing dependencies. Because of this, if a image needs to be built for Windows, it will first check for a ``Dockerfile_windows`` file instead of the default one. If there isn't one, it means that the same ``Dockerfile``can be used for both Windows and Linux.

All Windows images will be based on the image "mcr.microsoft.com/windows/servercore:ltsc2019". There are a couple of features that are needed from this image, especially powershell.

Adds a README explaining the image building process, including the Windows Container image building process.

**Which issue(s) this PR fixes**:

Fixes #77268
Depends On: #76838

**Special notes for your reviewer**:

**Does this PR introduce a user-facing change?**:

```release-note
Kubernetes E2E test image manifest lists now contain Windows images.
```

### CODE REVIEW

**Comment by PatrickLang** [comment]
/assign @listx
Can you review this one too? it's the next part after #76838

**Comment by claudiubelu** [comment]
Just FYI, this PR only really has 1 commit (the last one). The other commits are from part 1, and they're here for the sake of avoiding merge conflicts and things like that. They will dissapear once part 1 merges.

File: test/images/resource-consumer/consume-cpu/BUILD
Comment (listx) [inline test/images/resource-consumer/consume-cpu/BUILD]: I'm not a Bazel wizard, but is there a way to make deps select `golang.org/x/sys/windows:go_default_library` only for the windows platform, and the `bertimus9...` one for everything else? Maybe @ixdy knows how...?

File: test/images/resource-consumer/consume-cpu/consume_cpu.go
Comment (listx) [inline test/images/resource-consumer/consume-cpu/consume_cpu.go]: Can you add a comment for humans here for what system understands this magic string?

File: test/images/sample-apiserver/Makefile
Comment (listx) [inline test/images/sample-apiserver/Makefile]: Should it say `/go/bin/linux_arm` instead of `go/bin/linux_arm` here?

File: test/images/resource-consumer/consume-cpu/BUILD
Comment (ixdy) [reply test/images/resource-consumer/consume-cpu/BUILD]: I think that's what this is doing already. (also, I'm pretty sure this section was auto-generated by gazelle.)

Comment (ixdy) [reply test/images/resource-consumer/consume-cpu/BUILD]: I'm not sure why gazelle decided to elaborate all platforms instead of using the `//conditions:default` case, but alas, that's what it did.

File: test/images/resource-consumer/consume-cpu/consume_cpu.go
Comment (ixdy) [reply test/images/resource-consumer/consume-cpu/consume_cpu.go]: this is a fairly standard [Go build tag](https://golang.org/pkg/go/build/#hdr-Build_Constraints), so I'm not sure that additional comments are necessary here.

File: test/images/resource-consumer/consume-cpu/BUILD
Comment (claudiubelu) [reply test/images/resource-consumer/consume-cpu/BUILD]: Indeed, this was auto-generated, the bazel job was complaining about it, if it wasn't like this. I'm not a fan of how this looks either.

File: test/images/sample-apiserver/Makefile
Comment (claudiubelu) [reply test/images/sample-apiserver/Makefile]: Indeed, I think you're right, since the ``find`` above targets ``/go/bin``. I didn't notice before, since this was copy-paste from the ``Makefile``.

**Comment by claudiubelu** [comment]
Will have to add test for the ``resource-consumer`` ``consume-memory`` bits.

File: test/images/resource-consumer/consume-cpu/BUILD
Comment (listx) [reply test/images/resource-consumer/consume-cpu/BUILD]: Ah OK. I was just wondering if there was a less verbose way of doing the same thing. But as it is autogenerated, I think there's no need to discuss further

File: test/images/resource-consumer/consume-cpu/consume_cpu.go
Comment (listx) [reply test/images/resource-consumer/consume-cpu/consume_cpu.go]: TIL. ACK

**Comment by marosset** [comment]
milestone /v1.19

**Comment by marosset** [comment]
milestone /v1.19

**Comment by dims** [comment]
@claudiubelu ignore that job failure ... not a `Required` job and a known issue being worked on.

**Comment by jtslear** [comment]
Hello, @jtslear here from Bug Triage! This PR has not been updated for a long time, so I'd like to check on the status. The code freeze is starting June 25th and while there is still plenty of time, we want to ensure that each PR has a chance to be merged on time.

@luxas and @ixdy it would appear we are awaiting a review.

As the PR is tagged for 1.19, is it still planned for this release?

**Comment by ixdy** [comment]
I'm probably not the right person for this review anymore. @BenTheElder @spiffxp do you have any suggestions?

**Comment by BenTheElder** [comment]
No particular suggestions, but based on OWNERS files linked by the not
/cc mkumatag justagustus

**Comment by jtslear** [comment]
Hello @kubernetes/sig-testing-pr-reviews and @kubernetes/sig-windows-bugs, I've not seen any forward progress on this PR.  Should this still be in the plan for v1.19?

**Comment by claudiubelu** [comment]
> 
> 
> Hello @kubernetes/sig-testing-pr-reviews and @kubernetes/sig-windows-bugs, I've not seen any forward progress on this PR. Should this still be in the plan for v1.19?

Hopefully, yes.

For the most part, this PR is pretty straightforward, it just adds the Windows bits for a couple of images. Only resource-consumer is a bit different, since it adds a couple of different things and there is no active maintainer for the image, unfortunately (no active contributor with >= 20% LOC contributed). We could opt to have that in a separate PR, if  there are any discussions about it.

Let's try to get it approved by pinging the image owners. :)

echoserver:
    - @dims

jessie-dnsutils:
    - @dims
    - @mkumatag

nonroot:
    - @tallclair

redis:
    - @mkumatag
    - @rmmh

resource-consumer: has no active maintainers.

sample-apiserver:
    - dims

**Comment by claudiubelu** [comment]
I've split the resource-consumer bits to another, as I was suggesting: https://github.com/kubernetes/kubernetes/pull/92719

This should make reviewing this PR a lot easier.

**Comment by jtslear** [comment]
Due to reaching the code freeze deadline, I'm going to bump this to the next release.

/milestone v1.20

**Comment by yliaog** [comment]
@listx could you please help review? thanks.

**Comment by listx** [comment]
/lgtm
/approve

/hold

Remove hold when ready.

**Comment by immuzz** [comment]
SIG Backlog Review Meeting

- To be discussed at next WG-Test-Infra meeting @claudiubelu @marosset

**Comment by claudiubelu** [comment]
/assign @spiffxp 

Could you approve the ``build/dependencies.yaml`` changes?

**Comment by mkorbi** [comment]
Hello @claudiubelu 
This PR has been updated frequently, but it looks like it stuck/waiting for feedback, so I'd like to check what's the status. The code freeze is starting 12.11.2020 (about 3-4 weeks from now) and while there is still plenty of time, we want to ensure that each PR has a chance to be merged on time.
As the PR is tagged for 1.20, is it still planned for this release?

**Comment by claudiubelu** [comment]
> 
> 
> Hello @claudiubelu
> This PR has been updated frequently, but it looks like it stuck/waiting for feedback, so I'd like to check what's the status. The code freeze is starting 12.11.2020 (about 3-4 weeks from now) and while there is still plenty of time, we want to ensure that each PR has a chance to be merged on time.
> As the PR is tagged for 1.20, is it still planned for this release?

Yep. It's only waiting for approval for the ``build/dependencies.yaml`` changes, which is only 2 LOC.

File: test/images/sample-apiserver/Makefile
Comment (spiffxp) [inline test/images/sample-apiserver/Makefile:31]: for followup: all of these versions seem out of date

**Review by spiffxp — APPROVED** [review]
/approve
/lgtm

**Pull request merged**
